src/adapter/breakpoints.ts

```diff
@@ -553,30 +583,38 @@ export class BreakpointManager {
     if (thread) {
       await (this._launchBlocker = Promise.all([
         this._launchBlocker,
-        ...result.new.map(b => b.set(thread, false)),
+        ...result.new.map(b => b.set(thread)),
       ]));
     }
 
-    const dapBreakpoints = await Promise.all(result.list.map(b => b.toProvisionalDap()));
+    const dapBreakpoints = await Promise.all(result.list.map(b => b.toDap()));
     this._breakpointsStatisticsCalculator.registerBreakpoints(dapBreakpoints);
+
+    // In the next task after we send the response to the adapter, mark the
+    // breakpoints as having been set.
+    delay(0).then(() => result.new.forEach(bp => bp.markSetCompleted()));
```

Comment:
Looks like another place where a "post-event handler" type thing would be useful.